### ISSUE TITLE
change layer not in local machine

### ISSUE THREAD

**Issue by avsingh999** [publiclab/mapknitter#419]
### What happened just before the problem occurred
we can see there is no an option for change layer in localhost but
**In localhost:**
![pub6](https://user-images.githubusercontent.com/25536022/54666323-7a4d7500-4b0f-11e9-9fbc-1a8dcdd1b250.png)
**It must be like in Production:**
![pub7](https://user-images.githubusercontent.com/25536022/54666384-9e10bb00-4b0f-11e9-987a-32011542a201.png)

### Relevant URLs

https://localhost:3000

### PublicLab.org username
singhav
(to help reproduce the issue)

### Browser, version, and operating system

For bug reports, fill out the above template; for feature requests, you can delete the template.

**Comment by divyabaid16** [comment]
Can I work on this?

**Comment by jywarren** [comment]
Oh, we'd love the help! I wonder if the many changes recently have affected
the configuration of Leaflet Environmental Layers:
https://github.com/publiclab/leaflet-environmental-layers/

**Comment by divyabaid16** [comment]
I think many new layers were added some files have been moved from \dist to \src. So, public/lib/leaflet-environmental-layers is not synchronized with the [https://github.com/publiclab/leaflet-environmental-layers/](https://github.com/publiclab/leaflet-environmental-layers/) . I tried to make both of them equal but it was a part of ```.gitignore```

**Comment by sagarpreet-chadha** [comment]
Hey @igniteeng000 , 
So there is already a PR opened for this issue : https://github.com/publiclab/mapknitter/pull/466 by @divyabaid16 .
Kindly comment on the issue before working so that this does not happen 😄 .

So i think both of you are solving it right , it is weird that the `leaflet-hash` library is currently not present in `bower.json` .

So i think in order to maintain uniformity across all the projects , lets use this library instead : 
https://github.com/publiclab/leaflet-environmental-layers/blob/cb4b72824d0186bba1fa1bca1bdafff8a1742992/package.json#L46

and see this on how to use it : 
https://github.com/publiclab/leaflet-environmental-layers/blob/cb4b72824d0186bba1fa1bca1bdafff8a1742992/example/index.html#L432

So this library also adds the current layer in the URL as well 🗺 . See this for live example : 
https://publiclab.github.io/leaflet-environmental-layers/example/index.html#3/43.00/-46.26/BL2-MapK

@sashadev-sky ...thank you for reviewing the PR's 🙌  . 

@igniteeng000 , @divyabaid16  ...would you both like to collaborate in solving this one ? 
Thank you 😄 !

**Comment by sashadev-sky** [comment]
@sagarpreet-chadha I still would like to figure out why I have the layers on my screen already! Can you please confirm that your code is up to date and without the updates made by @divyabaid16, you don't have the layers panel on your screen?

**Comment by divyabaid16** [comment]
Yes, @sagarpreet-chadha I would like to collaborate with @igniteeng000 here :smile: 

I just pulled all the commits made and I still can't see any layers @sashadev-sky 
![Selection_123](https://user-images.githubusercontent.com/32747809/55781626-40491080-5ac8-11e9-8c5d-a41b5100df6f.png)

Thanks!

**Comment by sagarpreet-chadha** [comment]
I think i can solve the confusion here . @sashadev-sky , if you delete the libraries installed from **bower** which should be in `lib` folder and run `bower install` again . The layers button should disappear . 
Most probably , there `leaflet-hash` library was removed after you installed mapknitter project . 
Do let me know if this works 😄 . Thanks !

**Comment by sashadev-sky** [comment]
Yes you're right it is because of leaflet-hash from missing. From the blame view of `bower.json` It looks like [5 months](257f64921976f7782560e171783c04e06a7a54a5) you deleted `leaflet-hash` from it.  Was this intended?

**Comment by sashadev-sky** [comment]
@sagarpreet-chadha confirming if you re-add  `"leaflet-hash": "*"` to the bottom of `bower.json` the site works again. Also, in my case it is impossible to just generate a root page with no hash after it. So without leaflet hash I got an error message and no view. But I don't understand what that would have to do with the layers pane.

**Comment by sashadev-sky** [comment]
Oh i see I just re-read what you said above about the URL! I think people have been trouble getting mapknitter to run ([see here](https://github.com/publiclab/mapknitter/issues/397)), so let's just add the `"leaflet-hash": "*"` back in. Its just one line of code and its fixed. And then we can check in with that user if it's working for him now because his error looked exactly like mine when I deleted leaflet-hash. 

Then we can open new PR regarding adding in new basemaps with leaflet-fullhash and dropping leaflet-hash.  What do you think? @sagarpreet-chadha

**Comment by sagarpreet-chadha** [comment]
@sashadev-sky awesome :smile: !
Yes , I think it was intended --- most probably i was facing a similar error then : 

![image](https://user-images.githubusercontent.com/14952645/55786101-3d065280-5ad1-11e9-9008-68cb905b745f.png)

I guess as we are shifting to `Leaflet-fullhash` , lets completely remove `leaflet-hash`  by removing this 2 lines : 
https://github.com/publiclab/mapknitter/blob/05cb5593f4c1b6cb0e786046da1cf82117359489/app/views/maps/index.html.erb#L5
and
https://github.com/publiclab/mapknitter/blob/05cb5593f4c1b6cb0e786046da1cf82117359489/app/views/maps/index.html.erb#L108

What do you think ? We can do either ways though  :)

Yes https://github.com/publiclab/mapknitter/issues/397 seems to be related to this 🙌 !
